src/main/java/com/autotune/analyzer/recommendations/engine/DurationBasedRecommendationEngine.java

```diff
@@ -83,181 +446,106 @@ public HashMap<String, Recommendation> generateRecommendation(ContainerData cont
+                        timestampToExtract,
+                        AnalyzerConstants.ResourceSetting.requests,
+                        AnalyzerConstants.RecommendationItem.memory,
+                        notifications);
+
+                // Current CPU Limit
+                RecommendationConfigItem currentCpuLimit = getCurrentValue(   filteredResultsMap,
+                        timestampToExtract,
+                        AnalyzerConstants.ResourceSetting.limits,
+                        AnalyzerConstants.RecommendationItem.cpu,
+                        notifications);
+
+                // Current Memory Limit
+                RecommendationConfigItem currentMemLimit = getCurrentValue(   filteredResultsMap,
+                        timestampToExtract,
+                        AnalyzerConstants.ResourceSetting.limits,
+                        AnalyzerConstants.RecommendationItem.memory,
+                        notifications);
+
+                // Create an internal map to send data to populate
+                HashMap<String, RecommendationConfigItem> internalMapToPopulate = new HashMap<String, RecommendationConfigItem>();
+                // Add current values
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.CURRENT_CPU_REQUEST, currentCpuRequest);
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.CURRENT_CPU_LIMIT, currentCpuLimit);
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.CURRENT_MEMORY_REQUEST, currentMemRequest);
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.CURRENT_MEMORY_LIMIT, currentMemLimit);
+                // Add recommended values
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.RECOMMENDED_CPU_REQUEST, recommendationCpuRequest);
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.RECOMMENDED_CPU_LIMIT, recommendationCpuLimits);
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.RECOMMENDED_MEMORY_REQUEST, recommendationMemRequest);
+                internalMapToPopulate.put(AnalyzerConstants.RecommendationEngine.InternalConstants.RECOMMENDED_MEMORY_LIMIT, recommendationMemLimits);
+
+                // Call the populate method to validate and populate the recommendation object
+                boolean isSuccess = populateRecommendation(
+                        recPeriod,
+                        recommendation,
+                        notifications,
+                        internalMapToPopulate,
+                        numPods,
+                        hours
+                );
+
+                // Use success in future if you want to avoid sending the recommendations
+                if (!isSuccess) {
+                    // Handle the issue accordingly
```

Comment:
I think we need to add notifications for the rest as well and not have to resort to fail the API